### DISCUSSION TITLE
New Uno Net7.0 Project from template displays "There were deployment issues" when building as "(Package)" because Net6.0 DLL not found

### DISCUSSION THREAD

**Discussion by ToddGlodek** [unoplatform/uno#12154] [General]
I just upgraded to Visual Studio Version 17.5.5.  I feel like I was able to do this before the upgrade - but since the upgrade I see this build error.

When creating the a new project using the project templates, I experience a "There were deployment issues" when running in "packaged" mode.   The error reported is "Cannot copy file" and the file referenced is a Net6.0 file - even though it is a Net7.0 project.

This is what I did :

First run the Update templates - "dotnet new install Uno.Templates" and observe "Success: Uno.Templates::4.8.11 installed the following templates:"

Create a new project :
<img width="540" alt="newProject" src="https://user-images.githubusercontent.com/19764509/236014043-63e09652-24c5-4737-adcc-45a5ee2e5569.png">

Do not modify the project in any way.

Then, run the project in "(Package)" mode and observer the following :
<img width="640" alt="DeploymentIssue" src="https://user-images.githubusercontent.com/19764509/236014272-ee048329-9940-4d8e-b6ac-7a58662b4d82.png">

NOTE:  If you try to run this as "(Unpackaged)" - it will build, but it displays a different error at runtime.  I will document that separately (#12155).

**Comment by nickrandolph** [comment]
So two things:
- Firstly, the template's missing some attributes in the Windows.csproj that improve the VS tooling. These shouldn't cause the error you're seeing but without them, VS will claim there's an issue the first time you build and run the Windows project. Add these to fix this issue:
```
		<!-- 
		Defining the "Msix" ProjectCapability here allows the Single-project MSIX Packaging
		Tools extension to be activated for this project even if the Windows App SDK Nuget
		package has not yet been restored.
	-->
	<ItemGroup Condition="'$(DisableMsixProjectCapabilityAddedByProject)'!='true' and '$(EnableMsixTooling)'=='true'">
		<ProjectCapability Include="Msix"/>
	</ItemGroup>

	<!-- 
		Defining the "HasPackageAndPublishMenuAddedByProject" property here allows the Solution 
		Explorer "Package and Publish" context menu entry to be enabled for this project even if 
		the Windows App SDK Nuget package has not yet been restored.
	-->
	<PropertyGroup Condition="'$(DisableHasPackageAndPublishMenuAddedByProject)'!='true' and '$(EnableMsixTooling)'=='true'">
		<HasPackageAndPublishMenu>true</HasPackageAndPublishMenu>
	</PropertyGroup>

```

- Secondly, the error you're seeing is most likely due to a bad download of the nuget package. For some reason we've seen this a bit with that version of the SDK. Please try removing that nuget folder (ie .nuget\package\smicrosoft.windowsppasdk\1.2.230313.1), restart VS and rebuild project and let me know if that fixes the issue

**Reply by ToddGlodek** [reply]
I appreciate your looking into this - I was just documenting the issue in case other people ran into the problem as well.  But its totally awesome that you are on it so fast !!  

I did the following

1. Added the XML snippet above to the Windows.csproj 
2.  Deleted  ALL the folders from .nuget\package\  and restarted Visual Studio

Attempted to run (using the versions of the NuGets specified by the new project template) -  same error

3. Upgraded to the latest version of the NuGets.  - same error

It seems like the "Microsoft.WindowsAppSDK" & "BuildTools"  versions can be defined in two separate places.  What happens if they are not the same - does the  non-common/Windows one just override ??

In the "common/shared" project it looks like this :
```
	<PropertyGroup>
		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows')) or '$(EnableWindowsTargeting)' == 'true'">$(TargetFrameworks);net7.0-windows10.0.19041</TargetFrameworks>
		<TargetFrameworks>$(TargetFrameworks);net7.0;</TargetFrameworks>

		<!-- Ensures the .xr.xml files are generated in a proper layout folder -->
		<GenerateLibraryLayout>true</GenerateLibraryLayout>
	</PropertyGroup>


       <ItemGroup>
	        <PackageReference Include="Uno.Dsp.Tasks" Version="1.0.1" />
		<PackageReference Include="Uno.WinUI" Version="4.8.33" />
		<PackageReference Include="Uno.Resizetizer" Version="1.0.4" />
		<PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.0" />
		<PackageReference Include="Uno.Material.WinUI" Version="2.5.3" />
		<PackageReference Include="Uno.Extensions.Authentication.WinUI" Version="2.3.10" />
		<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
	</ItemGroup>

	<Choose>
		<When Condition="$(IsWinAppSdk)">
			<ItemGroup>
				<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.3.230331000" />
				<PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
			</ItemGroup>
```

In the WINDOWS specific project - it looks like this :

```
	<PropertyGroup>
		<OutputType>WinExe</OutputType>
		<TargetFramework>net7.0-windows10.0.19041.0</TargetFramework>
		<TargetPlatformMinVersion>10.0.18362.0</TargetPlatformMinVersion>
		<RootNamespace>UnoWinUIQuickStart</RootNamespace>
		<Platforms>x86;x64;arm64</Platforms>
		<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
		<PublishProfile>win10-$(Platform).pubxml</PublishProfile>
		<UseWinUI>true</UseWinUI>
		<EnableMsixTooling>true</EnableMsixTooling>
	</PropertyGroup>
	<PropertyGroup>
		<!-- Bundles the WinAppSDK binaries (Uncomment for unpackaged builds) -->
		<!-- <WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained> -->
		<!-- This bundles the .NET Core libraries (Uncomment for packaged builds)  -->
		<SelfContained>true</SelfContained>
	</PropertyGroup>

	<ItemGroup>
		<Content Include="Images\**" />
		<Manifest Include="$(ApplicationManifest)" />
	</ItemGroup>

	<ItemGroup>
		<PackageReference Include="Uno.Resizetizer" Version="1.0.4" />
		<PackageReference Include="Uno.WinUI" Version="4.8.33" />
		**<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.3.230331000" />
		<PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />**
		<PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.0" />
		<PackageReference Include="Uno.Material.WinUI" Version="2.5.3" />
		<PackageReference Include="Uno.Extensions.Authentication.WinUI" Version="2.3.10" />
		<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
		<PackageReference Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
		<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.8.33" />
	</ItemGroup>

```
